### DISCUSSION TITLE
Using a cyberbike as a controller

### DISCUSSION THREAD

**Discussion by GhostBranch** [NicoHood/Nintendo#49] [Q&A]
I bought a used cyberbike. This is a gamecube controller made for the wii for cycling games. It should connect through a gamecube controller. Unfortunately the cable is missing so i am trying to make it myself using this library. I manage to give power to the controller. But I am not able to read it yet, it says Error reading Gamecube controller. Is there someone willing to help?

**Comment by Skuzee** [comment]
Do you need help finding the correct pins on the bike, or do you need help with some sort of programming question. You shouldn't need this library if you just intend to use the bike with a wii. I can't find much about the bike or what port it has on it, but I can give you some hints to what wires the bike needs: The joycon protocol is only 1 Data wire + ground + 3.3v to power the controller. (and 5v to power rumble motor on normal controller). So if you can find the correct wires you can hook them up to a gamecube extension cable.

I just checked a normal gamecube controller and measured from pin to pin, maybe these readings will correlate and help you figure out which pin is which:


|Multimeter Black | Multimeter Red | Reading Ohms
|--- | --- | ---|
|GND | Data | 5.5k |
| | 3.3v |4.7k |
| | 5v  | OL|
| | | | 
|Data | GND | 3.8k |
| | 3.3v |750 |
| | 5v  | OL|
| | | | 
|3.3v | Data | 750k |
| | GND |3.1k |
| | 5v  | OL|
| | | | 
|5v | Data | 315k |
| | GND |310k |
| | 3.3v  |310k |

**Comment by GhostBranch** [comment]
Okay so my goal is to make it into a general purpose controller to use over different project. I tried connecting it to a wii with jumper wires, unfortunately I don't have a extensioncable yet. This didn't work, when connecting it to the arduino I managed to give power to the bike and reset the cyberbike programming with instructions from the manual, this tells me that the bike is still working. After that I tried connecting a real gamecube controller to a arduino to make sure i was connecting everything correctly. This also worked. When I tried to measure resistances of the bike to find out the pinout of the bike I got measurements that didn't really match your results. I was able to find out some of the pinouts of the bike by opening the bike.
![WhatsApp Image 2021-12-08 at 15 12 17 (1)](https://user-images.githubusercontent.com/57985395/145223109-a0fb403d-6c8a-491d-b7a2-bb879c4009c1.jpeg)
.
![WhatsApp Image 2021-12-08 at 15 12 17](ht
![WhatsApp Image 2021-12-08 at 15 12 16 (1)](https://user-images.githubusercontent.com/57985395/145223135-7789bfd7-cbdc-4b50-b0bd-8c840438ffbf.jpeg)
tps://user-images.githubusercontent.com/57985395/145223124-630496c8-2c4a-4f82-972e-ae8a90cf5275.jpeg)
![WhatsApp Image 2021-12-08 at 15 12 16](https://user-images.githubusercontent.com/57985395/145223146-b75e6034-8c14-4159-95a8-48b392157ec6.jpeg)
![WhatsApp Image 2021-12-08 at 15 12 17](https://user-images.githubusercontent.com/57985395/145223392-01099967-ec64-46b1-9be4-6e361de3f691.jpeg)

The two red ports on the bottom right(8,9) are GND ports aswell.
I don't know what is wrong here please help.

**Reply by Skuzee** [reply]
If you tried to connect the bike to the Wii and it didn't work, then I assume that you're missing something, or misinterpreting the wires, or the bike is broken. Maybe there was some sort of converter box in the cable that you're missing. a 9 pin plug seems like a lot of connections for a Gamecube controller, or even USB. Honestly this project would be easier if you had all the parts and a working setup. you could use a cheap data analyzer (I got one for $8) to monitor all the communication and it would give you hints as to what is actually happening. 

There is no real pictures of the cyber bike online, nor the connecting cable. There is at least 3 version I can find. 
- A black cyber bike (that might plug into the Gamecube controller port?)
- A white cyber bike that has a USB type B on the front.
- A white cyber bike with a 9 pin connector on the front (your bike)

So it looks like its possible that some bikes might use a USB protocol instead of going through the controller ports? There might be some sort of converter box in the cable that converts it to a Gamecube controller plug? or splits the cable into multiple plugs. Tracking down a cable, or picture of a cable might be worth the effort. Find someone who owns one. 

Internally, the existence of both DAT and DATA might hint that there are multiple communication protocols. The 3.5v GND DATA lines might be Gamecube controller protocol? you might be able to connect to them directly to access the controller buttons? The 3.5v DAT CLK GND lines might be a different protocol like i2c?

**Reply by Skuzee** [reply]
More info:

that plug looks like a 9 pin mini-din (tybe b?) connector:
https://www.kraydad.com/mini-din-9-pin-b-type-cables/

I found this photo which suggest there is no converter box, and that you should be able to wire up the bike to the gamecube cable if you have the correct pins: 
https://webimg.secondhandapp.at/w-i-mgl/5cefc1c4ee552b67a9ae544f

Might want to physically follow the wires, or use a meter to make sure you have the correct pins for DATA, GND, 3.5V pins on the front plug.

**Comment by GhostBranch** [comment]
Okay I wasn't able to get it to work with the library so I decide to open the bike. I think it should be easy no to connect the buttons to a arduino but I wasn't able to this yet and I want some advice before trying more and proceeding. As you can see I can indentify for what every single wire is used.But now I just need to know how to connect it correctly to a arduino.
![WhatsApp Image 2022-01-29 at 18 22 07](https://user-images.githubusercontent.com/57985395/151670879-61e6b6eb-0fd4-498b-aa8e-c9c42be8f8bf.jpeg)
![WhatsApp Image 2022-01-29 at 18 23 45](https://user-images.githubusercontent.com/57985395/151670881-805eb769-10cf-4ca6-8ca7-656f0e423f4e.jpeg)
g

**Reply by genoa65-netizen** [reply]
e possibile collegare  direttamente i cavi insieme per rottura scheda di collegamento bici wii? e come collegare i fili visto che ha 5 + 6 +4 d entrata.

**Reply by genoa65-netizen** [reply]
cyberbike nera

**Comment by genoa65-netizen** [comment]
e possibile collegare direttamente i cavi insieme per rottura scheda di collegamento bici wii? e come collegare i fili visto che ha 5 + 6 +4 d entrata.

**Reply by genoa65-netizen** [reply]
cyberbike nera
